lib/metadata.ts

```diff
@@ -94,6 +98,22 @@ const parseMetadata = async (idpMeta: string, validateOpts): Promise<Record<stri
 
         if (X509Certificate) {
           ret.thumbprint = thumbprint.calculate(X509Certificate);
+          /**
+           * new crypto.X509Certificate fails with the X509Certificate cert without
+           * -----BEGIN CERTIFICATE-----
+           * and
+           * -----END CERTIFICATE-----
+           */
+          if (X509Certificate.indexOf(BEGIN) != -1 && X509Certificate.indexOf(END) != -1) {
+            const { validTo } = new crypto.X509Certificate(X509Certificate);
+            ret.validTo = validTo;
+          } else {
+            /**
+             * Prefixing -----BEGIN CERTIFICATE----- and suffixing -----END CERTIFICATE-----
+             */
+            const { validTo } = new crypto.X509Certificate(`${BEGIN}\n${X509Certificate}${END}`);
```

Comment:
@ukrocks007 We can possibly handle the case where one block is missing and possibly avoid duplicate blocks here.

Reply:
done